### DISCUSSION TITLE
Where is the right place to have a discussion on ranked-choice voting?

### DISCUSSION THREAD

**Discussion by ion-oset** [ion-oset/developer-commons#31] [Meta]
(This conversation started in Slack. We're transferring it here because it's a broader conversation than the specific question. I'm doing it mostly on a post-by-post basis so people can reply to specific posts.)

@cwulfman 
> I'd like to start a discussion about ranked-choice voting; Where do you think that discussion should go?

@ion-oset 
> is this discussion more on the general theory side or on the practical side - i.e. focused on a deliverable?

~~The short answer is: in the Discussions of a proposed "Standards and References" repository. Details are still under discussion.~~

**Conclusion**: The answer is: the ["Standards & References" category](https://github.com/ion-oset/developer-commons/discussions/categories/standards-references) of this repository.

**Comment by cwulfman** [comment]
Your question about the scope of the discussion I'm proposing is a good one, and it points to a drawback to the communication system we're concocting: it isn't always clear what bucket a conversation should go in when the conversation begins, and very often the context changes as the discussion develops.  To use this RCV discussion as an example: the immediate context for the discussion is https://github.com/TrustTheVote-Project/NIST-1500-100-103-examples/issues/60: a specific task to create a test case featuring RCV: soon I'll have a draft of that test case (which I can make available for review in a couple of ways: as a pull request, or as a short-lived branch against which tickets can be posted).  The larger context, though, is bringing the ElectOS dev team up to speed on RCV and how it is supported in NIST 1500 (both 100 and 103).  To that end, I've written a wiki page (https://github.com/TrustTheVote-Project/NIST-1500-100-103-examples/wiki/ranked_choice), which might also merit discussion and/or review. If I were able to create discussions in the NIST-1500-100-103 examples repo, that would probably be the right place to root this discussion, but that consideration bleeds into a larger consideration about the nature of that repo and my suggestion that it be re-organized into at least two repos: a catalog/library of standards (a conversation about that is happening here: https://github.com/ion-oset/developer-commons/discussions/14) and a repository of examples and test cases.  The TL;DR answer: I think it belongs in a Discussion forum attached to the standards and references repo.

**Reply by ion-oset** [reply]
One note: when we get Standards and References set up, we should get the information in that wiki page into either the wiki in that repository, or into a more formal write-up in the documentation there.

**Comment by ion-oset** [comment]
@cwulfman 

> Your question about the scope of the discussion I'm proposing is a good one, and it points to a drawback to the communication system we're concocting: it isn't always clear what bucket a conversation should go in when the conversation begins, and very often the context changes as the discussion develops.

Yes, exactly.

> The TL;DR answer: I think it belongs in a Discussion forum attached to the standards and references repo.

My initial reaction to your question was either the Developer Commons or the Developer Internals (a subset of what is now in DAVS) forum. I was thinking that it might be better to have some specific place to discuss the standards but I was reluctant to propose a Yet Another forum. But your proposal broke the ice for me on that and if others agree I think that's a sufficiently important general topic to warrant it's own Discussions forum.

**Comment by ion-oset** [comment]
@cwulfman 

> If I were able to create discussions in the NIST-1500-100-103 examples repo, that would probably be the right place to root this discussion.

At this point I'm leaning pretty strongly **against** that. We should have the discussion under the Standards repository, even if it goes into specifics. When it gets cooked enough to warrant real changes to code or documentation it can end up as an issue on the appropriate repository.I think it's bad idea to have a discussion area in every repository, bordering on anti-pattern. It fragments discussion far too much and makes it really hard to find everything that was said.

@stratofax's experiment with BallotLab's Discussions drove that home for me this week. While having all the BallotMaker issues under one project was helpful, between email, Discussions on our general project repositories, Discussions on that specific project and issues, and Slack we had 5-6 places I had to look to keep up (even with Notifications). That's way too much. I believe we should keep the general Discussions forums to a small number, 2-3 tops. What we need to do is get those 2-3 straight.

**Reply by ion-oset** [reply]
There's an exception there for teams to have Discussion areas to do their planning. But even then we need to be careful to have that be organization and project planning not topics of "general interest". I think we should err on going to the general forums for most posts and using labels to manage things. That cuts down on the number of places to look dramatically, and labels are _much_ easier to change than the locations of posts.

Also worth noting is that on a technical level team Discussions seem to be more limited than repository Discussions. There are no categories, no reply-threads, and no filters (???) so the conversation flow is very linear. This is fine for immediate planning but less useful for a larger conversation.

**Comment by ion-oset** [comment]
After further discussion even three of forums can be too many. It also makes sense to have [Standards & References](https://github.com/ion-oset/developer-commons/discussions/categories/standards-references) be category in this repository, and that's what we're doing.

**Comment by windoverwater** [comment]
FYI - at least for VTP, project whitepapers seem to run aground on terminology more often than night.  Might it be possible to have one glossary page for ElectOS?

If not, each project may want to have a glossary so that it becomes a common thing to expect to find / do.

?

-sandy
